proxy/proxy_test.go

```diff
@@ -429,6 +431,61 @@ func TestConfigureBackendURL(t *testing.T) {
 	}
 }
 
+func TestUpstreamAppendCaCertToRootCa(t *testing.T) {
```

Comment:
I think we could make this a generic test helper which could then be used in the different relevant places (authenticators, mutators, proxy, authorizers)